usaspending_api/search/v2/views/search.py

```diff
@@ -297,29 +61,32 @@ def post(self, request):
 
         sort = json_request.get("sort", fields[0])
         if sort not in fields:
-            raise InvalidParameterException("Sort value not found in fields: {}".format(sort))
+            raise InvalidParameterException("Sort value '{}' not found in requested fields: {}".format(sort, fields))
 
         subawards_values = list(contract_subaward_mapping.keys()) + list(grant_subaward_mapping.keys())
         awards_values = list(award_contracts_mapping.keys()) + list(loan_award_mapping) + \
             list(non_loan_assistance_award_mapping.keys())
-        if (subawards and sort not in subawards_values) or (not subawards and sort not in awards_values):
-            raise InvalidParameterException("Sort value not found in award mappings: {}".format(sort))
+
+        msg = "Sort value '{}' not found in {{}} mappings: {{}}".format(sort)
+        if (not subawards and sort not in awards_values):
+            raise InvalidParameterException(msg.format("award", awards_values))
+        elif (subawards and sort not in subawards_values):
+            raise InvalidParameterException(msg.format("subaward", subawards_values))
 
         # build sql query filters
         if subawards:
-            # We do not use matviews for Subaward filtering, just the Subaward download filters
             queryset = subaward_filter(filters)
+            values = {'subaward_number', 'piid', 'fain', 'award_type'}
 
-            values = {'subaward_number', 'award__piid', 'award__fain', 'award_type'}
```

Comment:
subaward materialized view contains these fields, joins to the award table are not necessary